chem_spectra/lib/converter/fid/bruker.py

```diff
@@ -0,0 +1,102 @@
+import nmrglue as ng
+import numpy as np  # noqa: F401
+import os
+
+from chem_spectra.lib.converter.fid.base import FidBaseConverter
+from chem_spectra.lib.converter.share import parse_params, parse_solvent
+
+def search_brucker_processed(td):
+    try:
+        pdata_dir = find_dir(td, 'pdata')
+        target_dir = get_processed_data(pdata_dir)
+        return target_dir
+    except:     # noqa: E722
+        return False
+
+def find_dir(path, name):
+    for root, dirs, _ in os.walk(path):
+        if name in dirs:
+            return os.path.join(root, name)
+    return False
+
+def get_processed_data(path):
+    processed_dirs = False
+    for root, dirs, files in os.walk(path):
+        if (len(dirs) > 0):
+            processed_dirs = []
+            for dir_name in dirs:
+                dir_path = os.path.join(root, dir_name)
+                processed_dirs.append(dir_path)
+    return processed_dirs
+
+class FidHasBruckerProcessed:
+    def __init__(self, target_dir, params=False, fname=''):
+        self.params = parse_params(params)
+        self.data = self.__read(target_dir, fname)
```

Comment:
I would not call the __read function in the constructor. Preferably I would use it lazily in the getter. According to the principle avoid errors in the constructor under all circumstances.

```python
    __init__(....):
        self.params = parse_params(params)
        self.__target = (target_dir, fname)
        sef.__data = None

    @property
    def data(self):
        if self.__data is None: 
           self.__data = self.__read(*self.__target)
        return self.__data
```